internal/advertisement-operator/broadcaster.go

```diff
@@ -30,120 +29,82 @@ var (
 	log logr.Logger
 )
 
+type AdvertisementBroadcaster struct {
+	//TODO: transform these functions in methods to avoid all that parameters
```

Comment:
I think we can wait for this todo addressed in this PR. It should not be complex to achieve and I think it would make the code a bit simpler to follow...

Reply:
will be done in the next PR